themes/_core/js/one-page-checkout.js

```diff
@@ -0,0 +1,272 @@
+    return false;
+  }else{
+    console.log('all good');
+    return true;
+  }
+}
+
+$(document).ready(() => {
+  let lastSentEmail = '';
+  let loginEmailReady = false;
+  let loginPasswordReady = false;
+  let debounceTimeout;
+
+  $('#guest-form #field-email').on('input', function () {
+    const input = $(this);
+    clearTimeout(debounceTimeout);
+
+    debounceTimeout = setTimeout(() => {
+      const email = input.val().trim();
+
+      const isValidEmail = isValidEmail(email);
+
+      if (isValidEmail && email !== lastSentEmail) {
+        lastSentEmail = email;
+
+        prestashop.emit('GuestEmailEntered', {
+          detail: {
+            email,
+            timestamp: new Date(),
+          },
+        });
+        console.log('Event "emailEntered" throw with :', email);
+      }
+    }, 5000);
+  });
+
+  $('#login-form input#field-email').on('input', function () {
+    clearTimeout(debounceTimeout);
+
+    debounceTimeout = setTimeout(() => {
+      let email = $(this).val();
+
+
+      const isValidEmail = isValidEmail(email);
+
+      if (isValidEmail) {
+        loginEmailReady = email;
+        prestashop.emit('LoginEmailReady', {
+          detail: {
+            email,
+            timestamp: new Date(),
+          },
+        });
+        console.log('Event "LoginEmailReady" throw with :', email);
+        if(loginPasswordReady) {
+          prestashop.emit('LoginReady', {
+            detail: {
+              email,
+              timestamp: new Date(),
+            },
+          });
+          console.log('Event "LoginReady" throw with :', email);
+        }
+      }
+    }, 4000);
+  });
+
+  $('#login-form input#field-password').on('input', function () {
+    clearTimeout(debounceTimeout);
+
+    debounceTimeout = setTimeout(() => {
+      loginPasswordReady = true;
+      prestashop.emit('LoginPasswordReady', {
+        detail: {
+          timestamp: new Date(),
+        },
+      });
+      console.log('Event "LoginPasswordReady"');
+      if (loginEmailReady) {
+        prestashop.emit('LoginReady', {
+          detail: {
+            loginEmailReady,
+            timestamp: new Date(),
+          },
+        });
+        console.log('Event "LoginReady" send with :', loginEmailReady);
+      }
+    }, 5000);
+  });
+
+  $('#customer-form input').on('input', function () {
+    clearTimeout(debounceTimeout);
+
+    debounceTimeout = setTimeout(() => {
+     if(checkCustomerFormIsValid()){
+       prestashop.emit('CustomerFormReady',{detail: {timestamp: new Date()}});
+     } else {
+       console.log('form NOT ready');
+     }
+    }, 5000);
+  });
+
+});
+
+prestashop.on('PersonalInformationError', (event) => {
+  console.error('PersonalInformationError :', event);
+  const errorDiv = $('#personal_information_error');
+  errorDiv.html(event.detail.message);
+  errorDiv.show('fast');
+})
+
+prestashop.on('GuestEmailEntered', (event) => {
+  prestashop.email = event.detail.email;
```

Comment:
you could think of making an

OnePageCheckout object that is available through `window`, I can imagine that some of the developers would like to extend some of the features behind OPC and not modify the core files directly

this is very important